saleor/checkout/utils.py

```diff
@@ -1090,19 +1090,32 @@ def set_external_shipping_id(checkout: Checkout, app_shipping_id: str):
 
 
 def get_external_shipping_id(container: Union["Checkout", "Order"]):
-    metadata_object: ModelWithMetadata
+    metadata_object: ModelWithMetadata | None
     if isinstance(container, Checkout):
         metadata_object = get_checkout_metadata(container)
     else:
         metadata_object = container
+    if not metadata_object:
+        return None
     return metadata_object.get_value_from_private_metadata(PRIVATE_META_APP_SHIPPING_ID)
 
 
-def delete_external_shipping_id(checkout: Checkout, save: bool = False):
+def delete_external_shipping_id(checkout: Checkout, save: bool = False) -> bool:
     metadata = get_or_create_checkout_metadata(checkout)
-    metadata.delete_value_from_private_metadata(PRIVATE_META_APP_SHIPPING_ID)
-    if save:
+    field_deleted = metadata.delete_value_from_private_metadata(
+        PRIVATE_META_APP_SHIPPING_ID
+    )
+    if save and field_deleted:
         metadata.save(update_fields=["private_metadata"])
+    return field_deleted
+
+
+@allow_writer()
+def create_checkout_metadata(checkout: "Checkout"):
+    try:
+        return CheckoutMetadata.objects.create(checkout=checkout)
+    except IntegrityError:
+        return checkout.metadata_storage
```

Comment:
We should never catch `IntegrityError`s in our code. Why not use `get_or_create` instead?